api/views.py

```diff
@@ -40,9 +40,17 @@ class AssetViewSet(ModelViewSet):
     def get_queryset(self):
         user = self.request.user
         is_admin = self.request.user.is_staff
+        queryset = Asset.objects.all()
+        query_params = self.request.query_params
+
         if not is_admin:
-            return Asset.objects.filter(assigned_to=user)
-        return Asset.objects.all()
+            queryset = Asset.objects.filter(assigned_to=user)
+
+        if query_params.get('email'):
+            email = query_params['email']
```

Comment:
@JaphethCA Looks good but it would be nice to add validation for the email before using it in the query.
You can use the django email validator. You could do something like:
```
from django.core.validators import validate_email
from django.core.exceptions import ValidationError
from rest_framework import serializers

def get_queryset(self):
    .......
    email = query_params['email']
    try:
        validate_email(email)
    except ValidationError as error:
        raise serializers.ValidationError(error.message)
    queryset = Asset.objects.filter(assigned_to__email=email)
```

Reply:
Okay, thanks. Will work on that

Reply:
Done!